src/fidesops/graph/data_type.py

```diff
@@ -23,9 +23,12 @@ def to_value(self, other: Any) -> Optional[T]:
     def empty_value(self) -> T:
         """A value that represents `empty` in whatever way makes sense for type T"""
 
-    @abstractmethod
     def truncate(self, length: int, val: T) -> T:
         """Truncates value to given length"""
+        logger.warning(
+            f"Length truncation not supported for {T} data_type. Using original masked value instead for update query."
+        )
+        return val
 
```

Comment:
Ah right, thinking how I'd obtain type in the abstract base class here...

Reply:
looks like it'd be a bit convoluted https://stackoverflow.com/questions/57706180/generict-base-class-how-to-get-type-of-t-from-within-instance so I'm going with diff wording for now in this message

Reply:
FYI Dawn suggested instead of typename the much simpler
```
def truncate(self, length: int, val: T) -> T:
  typename = self.__class__.__name__ 
 ```
 
Which will give you e.g. "IntTypeConverter"

Reply:
ooo awesome idea thanks!